source/slang/slang-state-serialize.cpp

```diff
@@ -878,7 +878,9 @@ struct LoadContext
         {
             const auto& srcTranslationUnit = base.asRaw(srcTranslationUnits[i]);
 
-            int index = frontEndReq->addTranslationUnit(srcTranslationUnit.language);
+            // TODO(JS): We should probably serialize off the module name
+            // Passing in nullptr will just generate the module name
+            int index = frontEndReq->addTranslationUnit(srcTranslationUnit.language, nullptr);
```

Comment:
Agreed. The module name should almost certainly be part of the state that gets serialized here, and it probably also belongs in the IR container we serialize and then load with `-r`.